pkg/secure/mqttfactory.go

```diff
@@ -95,82 +83,31 @@ func (factory MqttFactory) Create(opts *mqtt.ClientOptions) (mqtt.Client, error)
 	return mqtt.NewClient(factory.opts), nil
 }
 
-func (factory MqttFactory) getSecrets() (*mqttSecrets, error) {
-	// No Auth? No Problem!...No secrets required.
-	if factory.authMode == AuthModeNone {
-		return nil, nil
-	}
-
-	secrets, err := factory.appContext.GetSecret(factory.secretPath)
-	if err != nil {
-		return nil, err
-	}
-	mqttSecrets := &mqttSecrets{
-		username:     secrets[MQTTSecretUsername],
-		password:     secrets[MQTTSecretPassword],
-		keyPemBlock:  []byte(secrets[MQTTSecretClientKey]),
-		certPemBlock: []byte(secrets[MQTTSecretClientCert]),
-		caPemBlock:   []byte(secrets[MQTTSecretCACert]),
-	}
-
-	return mqttSecrets, nil
-}
-
-func (factory MqttFactory) validateSecrets(secrets mqttSecrets) error {
-	caCertPool := x509.NewCertPool()
-	if factory.authMode == AuthModeUsernamePassword {
-		if secrets.username == "" || secrets.password == "" {
-			return errors.New("AuthModeUsernamePassword selected however Username or Password was not found at secret path")
-		}
-
-	} else if factory.authMode == AuthModeCert {
-		// need both to make a successful connection
-		if len(secrets.keyPemBlock) <= 0 || len(secrets.certPemBlock) <= 0 {
-			return errors.New("AuthModeCert selected however the key or cert PEM block was not found at secret path")
-		}
-	} else if factory.authMode == AuthModeCA {
-		if len(secrets.caPemBlock) <= 0 {
-			return errors.New("AuthModeCA selected however no PEM Block was found at secret path")
-		}
-	} else if factory.authMode != AuthModeNone {
-		return errors.New("Invalid AuthMode selected")
-	}
-
-	if len(secrets.caPemBlock) > 0 {
-		ok := caCertPool.AppendCertsFromPEM(secrets.caPemBlock)
-		if !ok {
-			return errors.New("Error parsing CA Certificate")
-		}
-	}
-
-	return nil
-}
-
-func (factory MqttFactory) configureMQTTClientForAuth(secrets mqttSecrets) error {
+func (factory MqttFactory) configureMQTTClientForAuth(secretData *messaging.SecretData) error {
 	var cert tls.Certificate
 	var err error
 	caCertPool := x509.NewCertPool()
 	tlsConfig := &tls.Config{
 		InsecureSkipVerify: factory.skipCertVerify,
 	}
 	switch factory.authMode {
-	case AuthModeUsernamePassword:
-		factory.opts.SetUsername(secrets.username)
-		factory.opts.SetPassword(secrets.password)
-	case AuthModeCert:
-		cert, err = tls.X509KeyPair(secrets.certPemBlock, secrets.keyPemBlock)
+	case messaging.AuthModeUsernamePassword:
+		factory.opts.SetUsername(secretData.Username)
+		factory.opts.SetPassword(secretData.Password)
+	case messaging.AuthModeCert:
+		cert, err = tls.X509KeyPair(secretData.CertPemBlock, secretData.KeyPemBlock)
 		if err != nil {
 			return err
 		}
 		tlsConfig.Certificates = []tls.Certificate{cert}
-	case AuthModeCA:
+	case messaging.AuthModeCA:
 		break
```

Comment:
remove the unnecessary break here; we could put a comment saying it's NOP and the reason for NOP